### PR TITLE
No default routes in v1alpha3

### PR DESCRIPTION
**Pull request by ijsnellf**
Removes default routing. VirtualServices must now have a route defined.

Also has some minor refactoring.

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v1/route.go [hunk 1]
```diff
@@ -415,17 +405,17 @@ func buildHTTPRouteV2(store model.IstioConfigStore, config model.Config, service
 // TODO: This logic is temporary until we fully switch from v1alpha1 to v1alpha2.
 // In v1alpha2, cluster names will be built using the subset name instead of labels.
 // This will allow us to remove this function, which is very inefficient.
-func fetchSubsetLabels(store model.IstioConfigStore, name, subsetName, domain string) (labels model.Labels) {
+func fetchSubsetLabels(store model.IstioConfigStore, fqdn, subsetName, domain string) (labels model.Labels) {
```
Comment (rshriram) [inline pilot/pkg/proxy/envoy/v1/route.go]: can we get rid of this function in subsequent PR?

Comment (ijsnellf) [reply pilot/pkg/proxy/envoy/v1/route.go]: I'd love to. It will require a major refactor of how we build or cluster names / clusters though.

**Comment by ldemailly** [comment]
please wait for a couple more reviews before merging this, thanks!

**Comment by ZackButcher** [comment]
@kyessenov mind taking a look too just to verify?

**Comment by andraxylia** [comment]
I see the default cluster has been removed in this PR. This is the opposite of what I needed for 503s. Can you elaborate on the need and justification for this change?

Does this mean that for every service we must configure a destination policy, otherwise there is no default cluster?

**Pull request merged**